### ISSUE TITLE
Batch size > 1 for quantized model gives `CUDA_ERROR_ILLEGAL_ADDRESS`

### ISSUE THREAD

**Issue by EricLBuehler** [EricLBuehler/mistral.rs#9]
If the number of choices is greater than 1 and the model is quantized, a `CUDA_ERROR_ILLEGAL_ADDRESS` error occurs. Please see huggingface/candle#1830.

```bash
thread '<unnamed>' panicked at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cudarc-0.10.0/src/driver/safe/core.rs:208:76:
called `Result::unwrap()` on an `Err` value: DriverError(CUDA_ERROR_ILLEGAL_ADDRESS, "an illegal memory access was encountered")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cudarc-0.10.0/src/driver/safe/core.rs:208:76:
called `Result::unwrap()` on an `Err` value: DriverError(CUDA_ERROR_ILLEGAL_ADDRESS, "an illegal memory access was encountered")
stack backtrace:
   0:     0x5620fca9acc6 - std::backtrace_rs::backtrace::libunwind::trace::hbee8a7973eeb6c93
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/../../backtrace/src/backtrace/libunwind.rs:104:5
   1:     0x5620fca9acc6 - std::backtrace_rs::backtrace::trace_unsynchronized::hc8ac75eea3aa6899
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x5620fca9acc6 - std::sys_common::backtrace::_print_fmt::hc7f3e3b5298b1083
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:68:5
   3:     0x5620fca9acc6 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hbb235daedd7c6190
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x5620fcacb430 - core::fmt::rt::Argument::fmt::h76c38a80d925a410
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/fmt/rt.rs:142:9
   5:     0x5620fcacb430 - core::fmt::write::h3ed6aeaa977c8e45
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/fmt/mod.rs:1120:17
   6:     0x5620fca9744f - std::io::Write::write_fmt::h78b18af5775fedb5
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/io/mod.rs:1810:15
   7:     0x5620fca9aaa4 - std::sys_common::backtrace::_print::h5d645a07e0fcfdbb
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x5620fca9aaa4 - std::sys_common::backtrace::print::h85035a511aafe7a8
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x5620fca9c327 - std::panicking::default_hook::{{closure}}::hcce8cea212785a25
  10:     0x5620fca9c089 - std::panicking::default_hook::hf5fcb0f213fe709a
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:292:9
  11:     0x5620fca9c7b8 - std::panicking::rust_panic_with_hook::h095fccf1dc9379ee
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:779:13
  12:     0x5620fca9c692 - std::panicking::begin_panic_handler::{{closure}}::h032ba12139b353db
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:657:13
  13:     0x5620fca9b1c6 - std::sys_common::backtrace::__rust_end_short_backtrace::h9259bc2ff8fd0f76
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:171:18
  14:     0x5620fca9c3f0 - rust_begin_unwind
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:645:5
  15:     0x5620fc114545 - core::panicking::panic_fmt::h784f20a50eaab275
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panicking.rs:72:14
  16:     0x5620fc114a93 - core::result::unwrap_failed::h03d8a5018196e1cd
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/result.rs:1649:5
  17:     0x5620fc2eef14 - <cudarc::driver::safe::core::CudaSlice<T> as core::ops::drop::Drop>::drop::h346fb06c9ea331ee
  18:     0x5620fc2fabd4 - alloc::sync::Arc<T,A>::drop_slow::h0ba9570eef3efe07
  19:     0x5620fc2fb780 - alloc::sync::Arc<T,A>::drop_slow::h967574ea0c28f711
  20:     0x5620fc22283c - <mistralrs_core::models::quantized_llama::Mlp as candle_core::Module>::forward::h86daf01e771f1fb0
  21:     0x5620fc222b0d - <mistralrs_core::models::quantized_llama::MlpOrMoe as candle_core::Module>::forward::heeb5f517c1c2e137
  22:     0x5620fc239d16 - mistralrs_core::models::quantized_llama::ModelWeights::forward::h099dfc92cebfb06d
  23:     0x5620fc28f151 - <mistralrs_core::pipeline::mistral::MistralPipeline as mistralrs_core::pipeline::Pipeline>::forward::h13f21c2f9558554a
  24:     0x5620fc2008f7 - mistralrs_core::engine::Engine::run::h621f9a92d3156163
  25:     0x5620fc1f50d3 - std::sys_common::backtrace::__rust_begin_short_backtrace::hdf757670f422fcf5
  26:     0x5620fc2b615c - core::ops::function::FnOnce::call_once{{vtable.shim}}::h0b1fd19c1138ec5f
  27:     0x5620fcaa1fc5 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h12de4fc57affb195
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/alloc/src/boxed.rs:2015:9
  28:     0x5620fcaa1fc5 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h3c619f45059d5cf1
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/alloc/src/boxed.rs:2015:9
  29:     0x5620fcaa1fc5 - std::sys::unix::thread::Thread::new::thread_start::hbac657605e4b7389
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys/unix/thread.rs:108:17
  30:     0x7fd040694ac3 - start_thread
                               at ./nptl/pthread_create.c:442:8
  31:     0x7fd040726a40 - clone3
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
  32:                0x0 - <unknown>
thread '<unnamed>' panicked at library/core/src/panicking.rs:163:5:
panic in a destructor during cleanup
thread caused non-unwinding panic. aborting.
Aborted (core dumped)
```

**Comment by EricLBuehler** [comment]
A minimum reproducible example with Candle [here](https://github.com/EricLBuehler/candle/blob/segfault/candle-examples/examples/quantized/main.rs), with [these 3 lines changed](https://github.com/EricLBuehler/candle/blob/3184030de6d4d7db91637e52e6a6e50beb9e1152/candle-examples/examples/quantized/main.rs#L499-L501). 

This shows that the error does not originate from the mistral.rs implementation.

**Comment by EricLBuehler** [comment]
Fixed in huggingface/candle#1832, awaiting merge.
